gsutil.py

```diff
@@ -73,24 +73,40 @@ def OutputAndExit(message):
     ('pyasn1-modules', ''),  # oauth2client dependency
     ('rsa', ''),  # oauth2client dependency
     ('apitools', ''),
-    ('boto', ''),
+    ('boto', ''), # XML API and AWS interop dependency
     ('gcs-oauth2-boto-plugin', ''),
     ('fasteners', ''), # oauth2client and apitools dependency
     ('monotonic', ''), # fasteners dependency
     ('httplib2', 'python2'),
     ('python-gflags', ''),
     ('retry-decorator', ''),
-    ('six', ''),
+    ('six', ''), # Python 2 / 3 compatibility dependency
     ('socksipy-branch', ''),
 ]
-for libdir, subdir in THIRD_PARTY_LIBS:
-  if not os.path.isdir(os.path.join(THIRD_PARTY_DIR, libdir)):
-    OutputAndExit(
-        'There is no %s library under the gsutil third-party directory (%s).\n'
-        'The gsutil command cannot work properly when installed this way.\n'
-        'Please re-install gsutil per the installation instructions.' % (
-            libdir, THIRD_PARTY_DIR))
-  sys.path.insert(0, os.path.join(THIRD_PARTY_DIR, libdir, subdir))
+
+# These are special third party libraries that live in
+# `gsutil/third_party/vendored`. These libraries have modifications specifically
+# for gsutil to operate smoothly that may not yet be present in the latest
+# release of the package maintainer's repositoy.
+VENDORED_THIRD_PARTY_LIBS = [
+    ('boto', 'vendored') # XML API and AWS interop dependency
+]
+
+
+def prepend_to_path(THIRD_PARTY_LIBS, THIRD_PARTY_DIR):
```

Comment:
Functions should be named `LikeThis()`, not `like_this()`

Also, parameters should be named `like_this`, not `LIKE_THIS`.

Also-also, I don't think we need to pass in `THIRD_PARTY_DIR` here, since it never changes; we can hard-code it into the function.

I think this is a bit more intuitive to read, thoughts?:
```
def PrependThirdPartyLibToPath(path_component_groups):
```

Reply:
Will update to PascalCase! And whoops my bad on the params

With Joe's comment I was gonna have `THIRD_PARTY_DIR` and `VENDOR_DIR` constants passed into `prepend_to_path` so we don't have to specify the `vendored` subdir every time we add a vendored lib.

Agreed that `path_component_groups` sounds nicer! Given above, how about:

```
def PrependThirdPartyLibToPath(path_component_groups, component_path):
```